### PR TITLE
Implementation of normmax activation and loss and budget (k-subsets) activation. 

### PR DESCRIPTION
**Pull request by andre-martins**
This PR adds two sparse transformations to the entmax package:
- alpha-normmax (described in Blondel et al., 2020).  
- k-subsets budget: this returns a sparse vector which sums to k; when k=1 it recovers sparsemax.  

Bisection algorithms are implemented in both cases. For alpha-normmax, the corresponding Fenchel-Young loss is also implemented.

The activations and losses pass all the tests.

### CODE REVIEW

File: entmax/__init__.py [hunk 1]
```diff
@@ -1,19 +1,25 @@
-__version__ = "1.3.dev0"
+__version__ = "1.4.dev0"
```
Comment (bpopeters) [inline entmax/__init__.py]: Actually, keep it as 1.3.dev0. The last release was 1.2. We only need to change the `__version__` immediately after we release a new version on pip.

File: entmax/root_finding.py [hunk 2]
```diff
@@ -133,6 +134,129 @@ def backward(cls, ctx, dY):
...
+    def backward(cls, ctx, dY):
+        Y, = ctx.saved_tensors
+
+        # tau = (size_one  + sum(score[active]) - budget) / size_active
+        # dtau / dscore = 1 / size_active
+        # J = Diag(s) - ss' / size_active, s indicator vector of active indices.
+
+        active = (Y > 0) & (Y < 1)
+        active_size = active.sum(ctx.dim)
+        v_hat = ((dY * active).sum(ctx.dim) /
```
Comment (bpopeters) [inline entmax/root_finding.py:252]: This is the problem line. @andre-martins @vene everyone okay with increasing the minimum version to 1.2 so this is fine?

File: entmax/root_finding.py [hunk 3]
```diff
@@ -133,6 +134,129 @@ def backward(cls, ctx, dY):
...
+        if not isinstance(budget, torch.Tensor):
+            budget = torch.tensor(budget, dtype=X.dtype, device=X.device)
+
+        budget_shape = list(X.shape)
+        budget_shape[dim] = 1
+        budget = budget.expand(*budget_shape)
+
+        ctx.budget = budget
+        ctx.dim = dim
+        d = X.shape[dim]
```
Comment (bpopeters) [inline entmax/root_finding.py]: unused variable

File: entmax/root_finding.py [hunk 4]
```diff
@@ -133,6 +134,129 @@ def backward(cls, ctx, dY):
...
+
+        max_val, _ = X.max(dim=dim, keepdim=True)
+        min_val, _ = X.min(dim=dim, keepdim=True)
+
+        # If tau = max_val - B/N, X-tau <= B/N, hence sum(max(0, X-tau)) <= B.
+        # If tau = min_val - B/N, X-tau >= B/N, hence sum(min(1, X-tau)) >= B.
+        tau_lo = min_val - budget / X.shape[dim]
+        tau_hi = max_val - budget / X.shape[dim]
+
+        f_lo = (torch.clamp(X - tau_lo, min=0, max=1).sum(dim).unsqueeze(dim)
```
Comment (bpopeters) [inline entmax/root_finding.py]: unused variable

File: entmax/root_finding.py [hunk 5]
```diff
@@ -133,6 +134,129 @@ def backward(cls, ctx, dY):
...
+
+        ctx.alpha = alpha
+        ctx.dim = dim
+        d = X.shape[dim]
+
+        max_val, _ = X.max(dim=dim, keepdim=True)
+        tau_lo = max_val - cls._gp(1, alpha)  # 1
+        tau_hi = max_val - cls._gp(1 / d, alpha)  # (1/d)**(alpha-1)
+
+        f_lo = (cls._p(X - tau_lo, alpha) ** alpha).sum(dim) - 1
```
Comment (bpopeters) [inline entmax/root_finding.py]: `f_lo` is unused. This is the same as on master, so this problem has been around for a while...

File: entmax/losses.py [hunk 6]
```diff
@@ -241,6 +260,35 @@ def entmax_bisect_loss(X, target, alpha=1.5, n_iter=50):
...
+    ----------
+    X : torch.Tensor, shape=(n_samples, n_classes)
+        The input 2D tensor of predicted scores
+
+    target : torch.LongTensor, shape=(n_samples,)
+        The ground truth labels, 0 <= target < n_classes.
+
+    alpha : float or torch.Tensor
+        Tensor of alpha parameters (> 1) to use for each row of X. If scalar
+        or python float, the same value is used for all rows.
```
Comment (bpopeters) [inline entmax/losses.py]: trailing whitespace
